core/src/main/java/org/phoenixctms/ctsms/security/reencrypt/FileReEncrypter.java

```diff
@@ -0,0 +1,59 @@
+package org.phoenixctms.ctsms.security.reencrypt;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.crypto.SecretKey;
+
+import org.phoenixctms.ctsms.enumeration.FileModule;
+import org.phoenixctms.ctsms.security.CipherStream;
+import org.phoenixctms.ctsms.security.CryptoUtil;
+import org.phoenixctms.ctsms.security.VerifyMD5InputStream;
+import org.phoenixctms.ctsms.util.CommonUtil;
+import org.phoenixctms.ctsms.util.CoreUtil;
+
+public abstract class FileReEncrypter<T> extends ReEncrypter<T> {
+
+	@Override
+	protected void reEncrypt(T item, SecretKey oldDepartmentKey, SecretKey newDepartmentKey) throws Exception {
+		InputStream plainTextStream = new VerifyMD5InputStream(
+				CryptoUtil.createDecryptionStream(getIv(item), oldDepartmentKey, CoreUtil.createFileServiceFileInputStream(getExternalFileName(item))), getMd5(item));
+		String newFileName = CoreUtil.createExternalFileName(CoreUtil.getExternalFileDirectoryPrefix(getModule(item)), getExternalFileName(item));
+		java.io.File externalFile = new java.io.File(CoreUtil.getFileServiceExternalFilename(newFileName));
+		CipherStream cipherStream = CryptoUtil.createEncryptionStream(newDepartmentKey, new FileOutputStream(externalFile));
+		int nRead;
+		long totalRead = 0;
+		byte[] block = new byte[CommonUtil.INPUTSTREAM_BUFFER_BLOCKSIZE];
+		try {
+			while ((nRead = plainTextStream.read(block, 0, block.length)) != -1) {
```

Comment:
IOUtils offers `copy` for this purpose.

Reply:
seems we have commons-io as dep already. but it's currently not used anywhere afaics